mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/api/persistence/metadata/MetadataTypeDescriptorResultJsonSerializer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.persistence.metadata;
+
+import org.mule.runtime.api.metadata.descriptor.ImmutableTypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.TypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.resolving.ImmutableMetadataResult;
+import org.mule.runtime.api.metadata.resolving.MetadataResult;
+import org.mule.runtime.extension.internal.persistence.metadata.dto.EntityMetadata;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * Serializer that can convert a {@link MetadataResult} of {@link TypeMetadataDescriptor} type into a readable and processable
+ * JSON representation and from a JSON {@link String} to an {@link MetadataResult} instance
+ *
+ * @since 1.0
+ */
+public class MetadataTypeDescriptorResultJsonSerializer extends AbstractMetadataResultJsonSerializer {
+
+  public MetadataTypeDescriptorResultJsonSerializer(boolean prettyPrint) {
+    super(prettyPrint);
+  }
+
+  @Override
+  public String serialize(MetadataResult result) {
+    return gson.toJson(new EntityMetadata(result));
+  }
+
+  @Override
+  public ImmutableMetadataResult<ImmutableTypeMetadataDescriptor> deserialize(String result) {
```

Comment:
We need to use the implementations instead of the interfaces because gson uses that information for the serialization process. Exposing our internal implementation is horrible and we need to find a workaround for this.